lib/call-cabal-project-to-nix.nix

```diff
@@ -228,9 +228,9 @@ let
                   then throw "${inputMap.${repoData.url}.rev} may not match ${repoData.ref} for ${repoData.url} use \"${repoData.url}/${repoData.ref}\" as the inputMap key if ${repoData.ref} is a branch or tag that points to ${inputMap.${repoData.url}.rev}."
                   else inputMap.${repoData.url})
             else if repoData.sha256 != null
-              then fetchgit { inherit (repoData) url sha256; rev = repoData.ref; }
+            then fetchgit { inherit (repoData) url sha256 rev; }
             else
-              let drv = builtins.fetchGit { inherit (repoData) url ref; };
+              let drv = builtins.fetchGit { inherit (repoData) url ; rev = repoData.rev or repoData.ref; ref = repoData.ref or null; };
               in __trace "WARNING: No sha256 found for source-repository-package ${repoData.url} ${repoData.ref} download may fail in restricted mode (hydra)"
```

Comment:
This `__trace` that uses `${repoData.ref}`, is that still safe? Should it use `${repoData.rev or repoData.ref}`?

Reply:
Good catch. I think a complication is that we may have a ref *and* a rev (which is part of what this is intending to allow). ie, I was trying to fetch a repo with no `master`, so it didn't fetch the right branch because `ref` was specified to be something else.

I'll update the trace message to include both.